Factory/src/main/java/org/example/notification/NotificationFactory.java

```diff
@@ -0,0 +1,39 @@
+package org.example.notification;
+
+import org.example.notification.enums.NotificationTypeEnum;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class NotificationFactory {
+
+    private static Map<NotificationTypeEnum, Notification> notifications;
+    private MailNotification mailNotification;
+    private SMSNotification smsNotification;
+
+    @Autowired
+    private NotificationFactory(MailNotification mailNotification, SMSNotification smsNotification) {
+        this.notifications = new HashMap<NotificationTypeEnum, Notification>() {{ }};
+        this.mailNotification = mailNotification;
+        this.smsNotification = smsNotification;
+    }
+
+    @PostConstruct
+    public void initFactory() {
+        notifications.put(NotificationTypeEnum.SMS, smsNotification);
+        notifications.put(NotificationTypeEnum.MAIL, mailNotification);
+    }
+
+    /**
+     * Return the notification type implementation
+     * @param type
+     * @return
+     */
+    public static Notification create(NotificationTypeEnum type) {
```

Comment:
We need to discuss better about this. I'm 100% sure that the pattern that you are using it is not the factory method. In fact you are using the simple factory. With factory method you will be able to have many types of creation methods.

Check these videos:
 https://www.youtube.com/watch?v=EcFVTgRHJLM&ab_channel=ChristopherOkhravi 
https://www.youtube.com/watch?v=EdFq_JIThqM&ab_channel=Geekific